sklearn/decomposition/_dict_learning.py

```diff
@@ -807,7 +794,7 @@ def dict_learning_online(X, n_components=2, *, alpha=1, n_iter=100,
     else:
         X_train = X
 
-    dictionary = check_array(dictionary.T, order='F', dtype=np.float64,
+    dictionary = check_array(dictionary, order='F', dtype=np.float64,
```

Comment:
It might be sufficient to call `asfortranarray` instead of making a full `check_array`
Maybe add the same comment regarding why using fortran as in the previous function.

Reply:
the initial dictionary can be provided by the user so it's better to use check_array. We should call check_array in dict_learning as well, but I'd rather do that in a separate PR where I refactor some parts (e.g. #18975).

Reply:
OK make sense then.